File: cpp/src/benchmark/phase/WarmupPhase.cpp
#include "WarmupPhase.h"
#include <benchmark/Benchmark.h>

#include <utility>

namespace Observatory {

WarmupPhase::WarmupPhase(Benchmark &benchmark, Operation &operation, uint32_t operationCount) :
        BenchmarkPhase(benchmark),
        operation(operation),
        operationCount(operationCount) {}

const char* WarmupPhase::getName() {
    return "WarmupPhase";
}

Status WarmupPhase::execute() {
    return operation.warmUp(operationCount);
}

}

File: java/benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/WarmUpPhase.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.result.Status;

class WarmUpPhase extends BenchmarkPhase {

    private final Operation operation;
    private final int iterationCount;

    WarmUpPhase(Connection connection, Operation operation, int iterationCount) {
        super(connection);

        this.operation = operation;
        this.iterationCount = iterationCount;
    }

    @Override
    Status execute() {
        return operation.warmUp(iterationCount);
    }
}
